### PR TITLE
fix test request timeout for shorter tests

### PR DESCRIPTION
**Pull request by almogdepaz**
better concurrency for sync
#443

### CODE REVIEW

File: sync/syncer.go [hunk 1]
```diff
@@ -222,17 +223,25 @@ func (s *Syncer) getLayerBlockIDs(index mesh.LayerID) (chan mesh.BlockID, error)
...
-		_, err := s.sendLayerIDsRequest(v, index, ch)
+		c, err := s.sendLayerIDsRequest(v, index)
 		if err != nil {
 			s.Error("could not fetch layer ", index, " block ids from peer ", v) // todo recover from this
 			return nil, err
 		} else {
 			reqCounter++
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
```
Comment (y0sher) [inline sync/syncer.go]: why defer and no just move it after ? defer dosen't give you anything here

Comment (almogdepaz) [reply sync/syncer.go]: yala sababa

File: sync/syncer.go [hunk 2]
```diff
@@ -222,17 +223,25 @@ func (s *Syncer) getLayerBlockIDs(index mesh.LayerID) (chan mesh.BlockID, error)
...
+
+	ch := make(chan []uint32, len(m))
+	wg := sync.WaitGroup{}
 	for _, v := range m {
-		_, err := s.sendLayerIDsRequest(v, index, ch)
+		c, err := s.sendLayerIDsRequest(v, index)
 		if err != nil {
 			s.Error("could not fetch layer ", index, " block ids from peer ", v) // todo recover from this
 			return nil, err
 		} else {
```
Comment (y0sher) [inline sync/syncer.go]: you don't need else here because you return.

File: sync/syncer.go [hunk 3]
```diff
@@ -222,17 +223,25 @@ func (s *Syncer) getLayerBlockIDs(index mesh.LayerID) (chan mesh.BlockID, error)
...
-	ch := make(chan []uint32)
+
+	ch := make(chan []uint32, len(m))
+	wg := sync.WaitGroup{}
 	for _, v := range m {
-		_, err := s.sendLayerIDsRequest(v, index, ch)
+		c, err := s.sendLayerIDsRequest(v, index)
 		if err != nil {
 			s.Error("could not fetch layer ", index, " block ids from peer ", v) // todo recover from this
 			return nil, err
```
Comment (y0sher) [inline sync/syncer.go]: maybe you should continue ?

**Review by y0sher — APPROVED** [review]

**Pull request merged**